UnityProject/Assets/Scripts/Core/Addressables/CatalogLoader.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AddressableAssets;
+using UnityEngine.AddressableAssets.ResourceLocators;
+using UnityEngine.ResourceManagement.AsyncOperations;
+using UnityEngine.ResourceManagement.ResourceLocations;
+
+public class CatalogLoader : MonoBehaviour
+{
+    public List<IResourceLocation> locations;
+
+    public void LoadCatalog(string catalogURL)
+    {
+        Debug.Log("loadCatalog");
+        AsyncOperationHandle<IResourceLocator> handle = Addressables.LoadContentCatalogAsync(catalogURL);
+        handle.Completed += loadCatalogsCompleted;
+    }
+    void loadCatalogsCompleted(AsyncOperationHandle<IResourceLocator> obj)
+    {
+        Debug.Log("loadCatalogsCompleted ==> " + obj.Status);
+        if (obj.Status == AsyncOperationStatus.Succeeded)
+        {
+            loadResourceLocation();
+        }
+        else
+        {
+            Debug.LogError("LoadCatalogsCompleted is failed");
+        }
+    }
+
+    void loadResourceLocation()
+    {
+        Debug.Log("loadResourceLocation");
+        var someList = new List<IResourceLocation>();
+        AsyncOperationHandle<IList<IResourceLocation>> handle = Addressables.LoadResourceLocationsAsync(someList);
+        
+        handle.Completed += locationsLoaded;
+    }
+    void locationsLoaded(AsyncOperationHandle<IList<IResourceLocation>> obj)
+    {
+        Debug.Log("locationsLoaded ==> " + obj.Status);
+        if (obj.Status == AsyncOperationStatus.Succeeded)
+        {
+            locations = new List<IResourceLocation>(obj.Result);
+            loadDependency();
+        }
+        else
+        {
+            Debug.LogError("locationsLoaded is failed");
```

Comment:
There's a bunch of `Debug.Log` calls in this file. Perhaps use `Logger.Log`, Category.Addressables`?

Reply:
yeah, that's from Danzer